packages/backend/src/services/action.ts

```diff
@@ -70,13 +75,16 @@ export const processAction = async (options: ProcessActionOptions) => {
         status: error.response.status,
         statusText: error.response.statusText,
       }
+      await sendErrorEmail(flow, $.userEmail, execution.testRun)
```

Comment:
Some refactoring points:
- if is CancelFlowError, terminate early (do not console.error)
- only call sendErrorEmail only once